### DISCUSSION TITLE
Standard library (glibc) dependencies

### DISCUSSION THREAD

**Discussion by HervisDaubeny** [borntyping/rust-simple_logger#73] [Support]
### Quick intro / context
Our team works with rust library for c2pa and we added your logger to the project. The metioned library has dependency on `glibc 2.28` so we run our application on Debian 11 (which has the library `glibc 2.31` by default). 
Latest version (4.0.0) of you logger however has dependecy on `glibc 2.34`, which is comparable to breaking change for us, since Deb11 can't go over `2.31` version.

### Question
I'd like to ask if any of your logger releases has dependency on `glibc 2.31` or lower so we could solve our problem just by using that version instead of the latest.

**Comment by borntyping** [comment]
I'm not sure if there's any dependency from `simple_logger` on glibc ([Cargo.toml](https://github.com/borntyping/rust-simple_logger/blob/main/Cargo.toml#L18-L21)). I think a few of the dependencies depend on the `libc` crate, but I'm not sure how the versioning there lines up.

You might be able to use `simple_logger` without the optional dependencies, if one of those is the one pulling in glibc/libc (see [Usage](https://github.com/borntyping/rust-simple_logger#usage) in the README file).

```toml
[dependencies.simple_logger]
default-features = false
```
